client.d.ts

```diff
@@ -0,0 +1,316 @@
+import { Socket } from 'socket.io-client';
+
+export type SocketIOFileUploadClientOptions =
+{
+  /**
+   * Define if data should be sent as a string or as a buffer.
+   * (`false` reads files as an octet array)
+   * @default false
+   */
+  useText: boolean
```

Comment:
Note that all values in the options bag are also properties on the client. From the docs:

> Each public property can be set up in an object passing at second parameter of the Siofu constructor:

Perhaps you could put it in the Options interface and then extend it below.

Reply:
Indeed, I completely missed that! 
Sadly, extending the interface would require redefining all properties of the inherited object. Instead, I believe it would be better to convert the `Options` interface into an `abstract class` and create a `Partial` type from the class properties.